include/matx/transforms/cub.h

```diff
@@ -1222,25 +1221,25 @@ void cub_sum(OutputTensor &a_out, const InputOperator &a,
 
 #ifndef MATX_DISABLE_CUB_CACHE
   auto params =
-      detail::matxCubPlan_t<OutputTensor, InputOperator,
+      detail::matxCubPlan_t<OutputTensor,
+                            InputOperator,
                             detail::CUB_OP_REDUCE_SUM>::GetCubParams(a_out, a, stream);
 
-
-  // Get cache or new Sort plan if it doesn't exist
-  auto ret = detail::cub_cache.Lookup(params);
-  if (ret == std::nullopt) {
-    auto tmp = new detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_SUM>{a_out, a, {}, stream};
-    tmp->ExecSum(a_out, a, stream);
-    detail::cub_cache.Insert(params, static_cast<void *>(tmp));
-  }
-  else {
-    auto type =
-        static_cast<detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_SUM> *>(ret.value());
-    type->ExecSum(a_out, a, stream);
-  }
+  using cache_val_type = detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_SUM, int>;
+  detail::GetCache().LookupAndExec<detail::cub_cache_t>(
+      detail::GetCacheIdFromType<detail::cub_cache_t>(),
+      params,
+      [&]() {
+        int test; ///\todo TYLER_TODO: this should not be needed
```

Comment:
What is this for?

Reply:
updated to remove int and use correct empty param based on our discussion